dataset/create_dataset.py

```diff
@@ -254,6 +255,24 @@ def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
         out_layer = out_mem_ds.CreateLayer('0', self.vec_srs, geom_type=ogr.wkbMultiPolygon)
         ogr.Layer.Clip(self.mem_vec_ds.GetLayer(), mem_layer, out_layer)
 
+        # Check if the feature geometry is polygonal:
+        feature_defn = out_layer.GetLayerDefn()
+        out_layer.ResetReading()
+        feature = out_layer.GetNextFeature()
+        while feature is not None:
+            geom = feature.GetGeometryRef()
+            name = geom.GetGeometryName()
+
+            # Approximate a curvature by a polygon geometry:
+            if name != "POLYGON" and name != "MULTIPOLYGON":
+                linear_geom = geom.GetLinearGeometry()
+                new_feature = ogr.Feature(feature_defn)
+                new_feature.SetGeometryDirectly(linear_geom)
+                out_layer.CreateFeature(new_feature)
+                out_layer.DeleteFeature(feature.GetFID())
+
+            feature = out_layer.GetNextFeature()
+
```

Comment:
Could this have been a "convert_to_polygons" function of it's own, called in \_\_getitem__, dealing with a layer as input and output?

Reply:
Done